modules/KalturaSupport/resources/webcast/cuePointsManager.js

```diff
@@ -0,0 +1,501 @@
+(function (mw, $) {
+    "use strict";
+    mw.webcast = mw.webcast || {};
+
+    mw.webcast.CuePointsManager = mw.KBasePlugin.extend({
+        /* DEVELOPER NOTICE: you should not set any property directly here (they will be shared between instances) - use the setup function instead */
+        setup: function () {
+            var _this = this;
+
+            /*
+             DEVELOPER NOTICE: you should set properties here (they will be scoped per instance)
+            */
+            $.extend(_this,{
+                _nextPendingCuePointIndex: 0,
+                _lastHandledServerTime: null,
+                _monitoredCuepoints : {
+                    entryContext : null, // this value is being initialized by function 'resetMonitorVariables'
+                    intervalId : null,
+                    tagsLike : '',
+                    enabled : false,
+                    typesMapping : {}
+                }
+            });
+
+            _this.resetMonitorVariables();
+            _this.addBindings();
+
+            _this.log('initialize(): invoked');
+        },
+        getCuePoints : function()
+        {
+            var player = this.getPlayer();
+            var cuePoints = (player && player.kCuePoints) ? player.kCuePoints.getCuePoints() : null;
+
+            return cuePoints || [];
+        },
+        addBindings : function()
+        {
+            var _this = this;
+
+            _this.log('addBindings(): registering to events with bind postfix ' + _this.bindPostFix);
+
+            _this.bind("onChangeMedia", function() {
+                _this.log('addBindings(onChangeMedia): stop the monitor process');
+                _this.stopMonitorProcess({reset : true});
+            });
+
+            _this.bind("playerReady", function() {
+
+                var player = _this.getPlayer();
+                var shouldRun = player && ((player.isLive() && mw.getConfig("EmbedPlayer.LiveCuepoints")) || player.kCuePoints);
```

Comment:
when player ready - verify if should handle cuepoints (either vod with cuepoints or live).

We need to perform this check each time 'playerReady' to support 'channel playlist'